### PR TITLE
Added date validation

### PR DESCRIPTION
**Pull request by FreezyEx**
Added Date validation to the Form class.
Valid syntax defined [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax)

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
Cool idea, it looks good on first sight, but I'll do a proper review tonight.

File: src/form.ts [hunk 1]
```diff
@@ -148,4 +148,25 @@ export class ConversationForm<C extends Context> {
...
+     * the user that they did not send a valid Date.
+     *
+     * @param otherwise Handler that will be run for skipped updates
+     * @returns The received date
+     */
+    async date(otherwise?: (ctx: C) => unknown | Promise<unknown>) {
+        const ctx = await this.conversation.wait();
+        const text = ctx.msg?.text ?? ctx.msg?.caption ?? "";
+        let date = new Date(text);
+        if(date === "Invalid Date") {
```
Comment (KnorpelSenf) [inline src/form.ts:166]: This comparison looks sketchy, though, are you sure that dates work like that?

Comment (FreezyEx) [reply src/form.ts]: Fixed converting `date `to a string

**Comment by KnorpelSenf** [comment]
Please revert that fix. You committed a pnpm lock file and reformatted the entire project.

If you do not have Deno installed or are unsure how to fix up CI, just drop me a comment and I can fix it up :)

**Comment by FreezyEx** [comment]
> Please revert that fix. You committed a pnpm lock file and reformatted the entire project.
> 
> If you do not have Deno installed or are unsure how to fix up CI, just drop me a comment and I can fix it up :)

Installed Deno and reverted. I cannot get format-and-lint to work without modifying the whole file with `deno fmt`

**Comment by KnorpelSenf** [comment]
I can fix it up later tody

**Comment by KnorpelSenf** [comment]
Oh nice, you got it already

**Comment by FreezyEx** [comment]
> Oh nice, you got it already

Do I need to do anything else?

**Comment by roziscoding** [comment]
> > Oh nice, you got it already
> 
> 
> 
> Do I need to do anything else?

it seems the job is still failing. @KnorpelSenf do you wanna take a look at it?

**Pull request closed without merging**